dev-tools/dependencies-report

```diff
@@ -34,7 +34,9 @@ done
 
 go mod tidy
 go mod download
-go list -m -json all $@ | go run go.elastic.co/go-licence-detector \
+GOPATH=`go env GOPATH`
+env GOBIN=$GOPATH/bin/ go install go.elastic.co/go-licence-detector@v0.4.0
```

Comment:
latest or 0.4.0 (the Makefile uses latest)? In Beats we try to pin the version. Additionally we have a [tools.go](https://github.com/elastic/beats/blob/master/tools/tools.go) (that is not build) to enforce the pinning and gives us an entry in the sum file (go modules behavior changes every minor release).

An older discussion about go modules and build tools: https://github.com/golang/go/issues/25922#issuecomment-413898264

Unfortunately go get/install tend to mess with the local go.sum and sometimes go.mod file. A project trying to workaround that is: https://github.com/bwplotka/bingo

Reply:
Would prefer to pin it.   Can switch both to 0.5.0.  Or can switch it to use tools.go.  Not loving this design because it generates licenses statements on a bunch of build tools that are not part of the build, and futher confuses go.mod.  Ideally the tools that we use to detect/generate licenses would be built outside the target tree.

This PR fails because the 7.x branch is not picking up the new go compiler.  This pr: https://github.com/elastic/fleet-server/pull/436 was pushed to master and backported, however the PR did not actually update the go version .  To fix that , this PR:  https://github.com/elastic/fleet-server/pull/346 was pushed to master... and for some reason not back ported.

Once we get the jenkins file fixed up, this PR should work.  We can then decided if go install is the wrong way to go.